sdks/go/pkg/beam/core/funcx/fn.go

```diff
@@ -631,6 +647,8 @@ func nextRetState(cur retState, transition ReturnKind) (retState, error) {
 		return -1, errEventTimeRetPrecedence
 	case RetValue, RetRTracker:
 		return rsOutput, nil
+	case RetProcessContinuation:
+		return rsProcessContinuation, nil
```

Comment:
From a technical standpoint this LGTM, I'm not so sure about the ordering of the changes though. Specifically, what I'm worried about is the potential for:

1) This PR takes away the validation that says you can't have a ProcessContinuation type as a return type.
2) The Sdk gets cut/released with version X.
3) This PR gets completed and we doc the change for version Y.
4) A user still on version X sees the doc changes and tries to use the ProcessContinuation, getting an error at pipeline run time (or even worse, silently seeing no errors)

While I'm 100% all about breaking PRs up, as a rule I would prefer we don't expose the user facing API until the last incremental PR so that we avoid the potential for something like that happening. Basically, I don't want to _expose_ any incomplete features.

I would prefer we get rid of these 2 lines (or at least comment them out) so that there is no ambiguity about what is supported and users get a pipeline construction time error indicating exactly what the problem is.

Reply:
That's good with me - I think its just these 2 lines we'd need to worry about, right?

Reply:
I think these two lines and the unit tests? I'll comment them out before the merge but leave them intact while we're in review so we are confident that any changes don't break anything

Reply:
That's fine by me - other than that I'm good with taking this

Reply:
Jack and I talked about this in our 1:1 and I'd prefer avoiding committing commented out code.

1. The feature isn't very discoverable without associated docs, so there's little risk in users trying it.
2. We can commit this first if we like, as long as we also change it up to return an error when we detect a ProcessContinuation is going to be required.  That way we can return a well formed, and clear error that can link to the jira explaining that this is an in progress feature, that is not yet available. 

This avoids awkward code that's commented out, and helps direct users to the right place to keep an eye on the feature if they want it, and further, we can test that error occurs explicitly in a test in the beam package, and here.

Reply:
> The feature isn't very discoverable without associated docs, so there's little risk in users trying it.

FWIW, the concern is more about users who are pinned to this release but see the doc update at a later time (since most sdk consumers aren't going to immediately update to the latest release)

> We can commit this first if we like, as long as we also change it up to return an error when we detect a ProcessContinuation is going to be required. That way we can return a well formed, and clear error that can link to the jira explaining that this is an in progress feature, that is not yet available.

That's good with me, agreed that its a nicer approach. I'm satisfied as long as we're not silently continuing or failing at runtime

Reply:
Returned an error when validating the ordering and updated the unit test to expect an error instead of an output Fn struct.